lib/src/circular_slider_paint.dart

```diff
@@ -124,16 +238,46 @@ class _CircularSliderState extends State<CircularSliderPaint> {
 
     var angle = coordinatesToRadians(_painter.center, position);
     var percentage = radiansToPercentage(angle);
-    var newValue = percentageToValue(percentage, widget.intervals);
+    var newValue = percentageToValue(percentage, widget.divisions);
 
+    if (isBothHandlersSelected) {
+      var newValueInit =
+          (newValue - _differenceFromInitPoint) % widget.divisions;
+      if (newValueInit != widget.init) {
+        var newValueEnd =
+            (widget.end + (newValueInit - widget.init)) % widget.divisions;
+        widget.onSelectionChange(newValueInit, newValueEnd, _laps);
+      }
+      return;
+    }
+
+    if (isSingleHandler) {
+      widget.onSelectionChange(widget.init, newValue, _laps);
+      return;
+    }
+
+    // isDoubleHandler but one handler was selected
     if (_isInitHandlerSelected) {
-      widget.onSelectionChange(newValue, widget.end);
+      widget.onSelectionChange(newValue, widget.end, _laps);
     } else {
-      widget.onSelectionChange(widget.init, newValue);
+      widget.onSelectionChange(widget.init, newValue, _laps);
     }
   }
 
-  void _onPanEnd(_) {
+  void _onPanEnd(Offset details) {
+    RenderBox renderBox = context.findRenderObject();
+    var position = renderBox.globalToLocal(details);
+    
+    var angle = coordinatesToRadians(_painter.center, position);
+    var percentage = radiansToPercentage(angle);
+    var newValue = percentageToValue(percentage, widget.divisions);
+
+    if (_isInitHandlerSelected) {
+      widget.onSelectionEnd(newValue, widget.end);
+    } else {
+      widget.onSelectionEnd(widget.init, newValue);
+    }
```

Comment:
this is a bit more complex now
we could have both handlers selected (click on the selection and dragging moves the whole selection) so we would need to account for that (check lines 243 to 251 in this same file for reference)
also we would need to return also the laps (same lines)